xrpl4j-crypto-parent/xrpl4j-crypto-core/src/main/java/org/xrpl/xrpl4j/crypto/core/signing/Signature.java

```diff
@@ -12,8 +12,8 @@
  * Represents a digital signature for a transaction that can be submitted to the XRP Ledger.
  */
 @Value.Immutable
-@JsonSerialize(as = ImmutableSignature.class)
-@JsonDeserialize(as = ImmutableSignature.class)
+@JsonSerialize(using = SignatureSerializer.class)
+@JsonDeserialize(using = SignatureDeserializer.class)
```

Comment:
These will probably get removed once you add the serializer/deserializer to the Xrpl4jModule, but for future reference, you need both the `as` and the `using` fields in these annotations for things to work correctly with immutables.

Reply:
https://github.com/XRPLF/xrpl4j/pull/303#discussion_r984657149

Reply:
This will be fixed via #306.